alot/commands/thread.py

```diff
@@ -934,72 +934,130 @@ async def apply(self, ui):
+        if not entry:
+            raise RuntimeError(
+                f'no mailcap handler found for MIME type {mimetype}')
+
+        afterwards = None  # callback, will rm tempfile if used
+        handler_stdin = None
+        tempfile_name = None
+        handler_raw_commandstring = entry['view']
+        # read parameter
+        part = attachment.get_mime_representation()
+        parms = tuple('='.join(p) for p in part.get_params())
+
+        # in case the mailcap defined command contains no '%s',
+        # we pipe the files content to the handling command via stdin
+        if '%s' in handler_raw_commandstring:
+            nametemplate = entry.get('nametemplate', '%s')
+            prefix, suffix = parse_mailcap_nametemplate(nametemplate)
+            tempfile_name, afterwards = \
+                self._make_temp_file(attachment, prefix, suffix)
+        else:
+            handler_stdin = BytesIO()
+            attachment.write(handler_stdin)
 
-            # create handler command list
-            handler_cmd = mailcap.subst(handler_raw_commandstring, mimetype,
-                                        filename=tempfile_name, plist=parms)
+        # create handler command list
+        handler_cmd = mailcap.subst(handler_raw_commandstring, mimetype,
+                                    filename=tempfile_name, plist=parms)
 
-            handler_cmdlist = split_commandstring(handler_cmd)
+        handler_cmdlist = split_commandstring(handler_cmd)
 
-            # 'needsterminal' makes handler overtake the terminal
-            # XXX: could this be repalced with "'needsterminal' not in entry"?
-            overtakes = entry.get('needsterminal') is None
+        # 'needsterminal' makes handler overtake the terminal
+        # XXX: could this be replaced with "'needsterminal' not in entry"?
+        overtakes = entry.get('needsterminal') is None
 
-            await ui.apply_command(ExternalCommand(handler_cmdlist,
-                                                   stdin=handler_stdin,
-                                                   on_success=afterwards,
-                                                   thread=overtakes))
+        return ExternalCommand(handler_cmdlist,
+                               stdin=handler_stdin,
+                               on_exit=afterwards,
+                               thread=overtakes,
+                               spawn=self.spawn)
+
+    @staticmethod
+    def _get_attachment(ui):
+        focus = ui.get_deep_focus()
+        if isinstance(focus, AttachmentWidget):
+            return focus.get_attachment()
+        elif (getattr(focus, 'mimepart', False) and
+              isinstance(focus.mimepart, Attachment)):
+            return focus.mimepart
         else:
-            ui.notify('unknown mime type')
+            raise RuntimeError('not focused on an attachment')
+
+    @staticmethod
+    def _make_temp_file(attachment, prefix='', suffix=''):
```

Comment:
The ExternalCommand in charge of executing the command using the temporary file does not accept context managers. It's not either possible to use a "with" block on top of it because it can spawn an independant thread.